diffrax/_step_size_controller/adaptive.py

```diff
@@ -353,35 +349,14 @@ def dynamics(t, y, args):
     dtmin: Optional[RealScalarLike] = None
     dtmax: Optional[RealScalarLike] = None
     force_dtmin: bool = True
-    step_ts: Optional[Real[Array, " steps"]] = eqx.field(
-        default=None, converter=_none_or_array
-    )
-    jump_ts: Optional[Real[Array, " jumps"]] = eqx.field(
-        default=None, converter=_none_or_array
-    )
```

Comment:
So I think it's important that we don't break the existing `PIDController`. If at all possible I really try to avoid backward-incompatible changes.

In this case I think we have two choices:
1. Keep all of the code here, but if `self.step_ts is not None` or `self.jump_ts is not None`, then emit a warning in `__check_init__` to prefer the new implementation instead.
2. Add a custom metaclass with a custom `__call__`, so that if `step_ts` or `jump_ts` is passed, then in fact we'll automatically rewrap them into the new implementation. (That is, `PIDController(..., jump_ts=...)` will actually return a `JumpStepWrapper(PIDController(...), ...)` ! Python allows for some crazy things.)

Reply:
Yes, I was just about to ask your advice on how to achieve the second point 😁.
Will do!